src/AutoRest.CSharp/Common/Generation/Writers/PageableMethodsWriterExtensions.cs

```diff
@@ -118,23 +118,51 @@ public static CodeWriter WritePageableBody(this CodeWriter writer, IReadOnlyList
             return writer.Line($"return {typeof(PageableHelpers)}.{(async ? nameof(PageableHelpers.CreateAsyncPageable) : nameof(PageableHelpers.CreatePageable))}({createPageableParameters.Join(", ")});");
         }
 
-        private static bool EnsureRequestContextVariable(this CodeWriter writer, IReadOnlyList<Parameter> parameters)
+        private static void AddTrailingPageableParameters(this List<FormattableString> createPageableParameters, IReadOnlyCollection<Parameter> methodParameters, string scopeName, string? itemPropertyName, string? nextLinkPropertyName)
+        {
+            createPageableParameters.Add($"{scopeName:L}");
+            createPageableParameters.Add($"{itemPropertyName:L}");
+            createPageableParameters.Add($"{nextLinkPropertyName:L}");
+
+            if (ContainsRequestContext(methodParameters))
+            {
+                createPageableParameters.Add($"{KnownParameters.RequestContext.Name:I}");
+            }
+            else if (methodParameters.Contains(KnownParameters.CancellationTokenParameter))
+            {
+                createPageableParameters.Add($"{KnownParameters.CancellationTokenParameter.Name:I}");
+            }
+        }
+
+        private static bool EnsureRequestContextVariable(this CodeWriter writer, MethodSignature methodSignature, RestClientMethod? createFirstPageRequestMethod, RestClientMethod? createNextPageRequestMethod)
         {
-            if (parameters.Contains(KnownParameters.RequestContext))
+            if (ContainsRequestContext(methodSignature.Parameters))
             {
-                return true;
+                return false;
```

Comment:
Why we changed to return `false` now? Previous codes:
> if (parameters.Contains(KnownParameters.RequestContext))
            {
                return true;
            }

Reply:
The usage of the method has changed. It now returns `true` if `RequestContext` variable must be added to the list of parameters (as if it was a parameter in the method signature). Then it can be used in `CreateXxxRequest` methods and in pageable itself